File: week2/week2-problem_sets/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);
int get_grade(float letters, float words, float sentences);

int main(void){
    string text = get_string("Text: ");
    int letters = count_letters(text);
    int words = count_words(text);
    int sentences = count_sentences(text);
    int grade = get_grade(letters, words, sentences);
}























int count_letters(string text){
    int length = strlen(text);
    int letters = 0;
    
    //calculating letters
    for(int i = 0; i < length; i++){
        if (isalpha(text[i])){
            letters++;
        }
    }
    return letters;
}

int count_words(string text){
    int length = strlen(text);
    int words = 1;
   
   //calculating words
    for(int i = 0; i < length; i++){
        if(isspace(text[i])){
            words++;
        }
    }
    return words;
}

int count_sentences(string text){
    int length = strlen(text);
    int sentences = 0;
   
   //calculating sentences
    for(int i = 0; i < length; i++){
        if(text[i] == '!' || text[i] == '?' || text[i] == '.'){
            sentences++;
        }
    }
    return sentences;
}

int get_grade(float letters, float words, float sentences){
    float L = (letters/words) * 100;
    float S = (sentences/words) * 100;
    int index = round((0.0588 * L) - (0.296 * S) - 15.8);
    if (index >= 16){
        printf("Grade 16+\n");
    }
    else if(index < 1){
        printf("Before Grade 1\n");
    }
    else {
      printf("Grade %i\n", index);  
    }
    return index;
}

File: week6/week6-problem_set/sentimental-readability/readability.py
def main():
    text = input("Text: ")
    letters = count_letters(text)
    words = count_words(text)
    sentences = count_sentences(text)
    grade = get_grade(letters, words, sentences)


def count_letters(text):
    length = len(text)
    letters = 0
    i = 0
    while (i < length):
        if text[i].isalpha() == True:
            letters += 1
        i += 1
    return letters

def count_words(text):
    length = len(text)
    words = 1
    i = 0
    while (i < length):
        if text[i].isspace() == True:
            words += 1        
        i += 1
    return words 

def count_sentences(text):
    length = len(text)
    sentences = 0
    i = 0
    while (i < length):
        if(text[i] == "!" or text[i] == "." or text[i] == "?"):
            sentences += 1
        i += 1
    return sentences

def get_grade(letters, words, sentences):
    L = (letters/words) * 100
    S = (sentences/words) * 100
    index = round((0.0588 * L) - (0.296 * S) - 15.8)
    if index >= 16:
        print("Grade 16+")
    elif index < 1:
        print("Before Grade 1")
    else:
        print(f"Grade {index}")
    return index

main()